var/spack/repos/builtin/packages/cgal/package.py

```diff
@@ -59,28 +65,53 @@ class Cgal(CMakePackage):
     depends_on('qt@5:', when='+demos')
 
     # Optional Third Party Libraries
+    depends_on('eigen', when='+eigen')
+
     # depends_on('leda')
     # depends_on('mpfi')
     # depends_on('rs')
     # depends_on('rs3')
     # depends_on('ntl')
-    # depends_on('eigen')
     # depends_on('libqglviewer')
     # depends_on('esbtl')
     # depends_on('intel-tbb')
 
+    conflicts('~header_only', when='@:4.9',
+              msg="Header only builds became optional in 4.9,"
+                  "default thereafter")
```

Comment:
```suggestion
                  " default thereafter")
```
You're missing a space in the help message.